src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1065,11 +1065,19 @@ def config_diagnostics(resource_group_name, name, level=None,
         application_logs = ApplicationLogsConfig(fs_log)
 
     http_logs = None
-    if web_server_logging is not None:
-        enabled = web_server_logging
-        # 100 mb max log size, retenting last 3 days. Yes we hard code it, portal does too
-        fs_server_log = FileSystemHttpLogsConfig(100, 3, enabled)
-        http_logs = HttpLogsConfig(fs_server_log)
+    logs_enabled = (web_server_logging if web_server_logging is not None
+                    else docker_container_logging)
+
+    # 100 mb max log size, retenting last 3 days. Yes we hard code it, portal does too
+    filesystem_log_config = FileSystemHttpLogsConfig(100, 3, logs_enabled == 'filesystem')
+    az_blob_storage_log_config = AzureBlobStorageHttpLogsConfig(None, 3, False)
```

Comment:
Should this be part of the `else` in the following control flow?

Reply:
The line containing filesystem_log_config I believe is fine. As for az_blob_storage_log_config, it will be overridden if ln 1078 (az_blob_storage_log_config = .. 3, True) gets executed. Either way it need to be included in http_logs. Using else will spawn an extra 2 lines of code and my way has unnecessary object creation (of a small object) if ln 1078 gets executed. Either one is fine in my opinion.